device/core/src/module_client.ts

```diff
@@ -178,7 +258,7 @@ export class ModuleClient extends InternalClient {
     }
 
     /*Codes_SRS_NODE_MODULE_CLIENT_05_006: [The fromConnectionString method shall return a new instance of the Client object, as by a call to new Client(new transportCtor(...)).]*/
-    return new ModuleClient(new transportCtor(authenticationProvider), null);
+    return new ModuleClient(new transportCtor(authenticationProvider), new MethodClient(authenticationProvider));
```

Comment:
Note: it's technically possible to have an EdgeModule that doesn't use fromEnvironment().  I'm using this for my E2E test work because it's way easier to debug a module outside of a container.  Whether or not we allow/support this scenario is another question.  

---
In reply to: [197231663](https://github.com/Azure/azure-iot-sdk-node/pull/322#discussion_r197231663) [](ancestors = 197231663)

Reply:
yeah it is possible and that's why i'm creating the MethodClient everywhere. which is also why it's never supposed to be null (see your earlier comment) or that'd be a bug. I feel it's simpler that way for now, until we get a good way to identify whether we're connecting to an Edge gateway or not... I'm not sure what we can do about it. One way would be to lazily initialize it when invokeMethod is called maybe? but then we have to check for its existence everywhere (setOptions etc)

---
In reply to: [197233150](https://github.com/Azure/azure-iot-sdk-node/pull/322#discussion_r197233150) [](ancestors = 197233150,197231663)